src/clj_test_containers/core.clj

```diff
@@ -179,13 +183,26 @@
      :stdout (.getStdout result)
      :stderr (.getStderr result)}))
 
+(defn dump-logs
+  [container-config]
+  ((:logs container-config)))
+
+(defn capture-logs
+  [container]
+  (let [to-string-consumer (ToStringConsumer.)]
+    (.followOutput container to-string-consumer)
+    (fn []
+      (-> (.toUtf8String to-string-consumer)
+          (clojure.string/replace #"\n+" "\n")))))
+
 (defn start!
   "Starts the underlying testcontainer instance and adds new values to the response map, e.g. :id and :first-mapped-port"
   [container-config]
-  (let [container (:container container-config)]
+  (let [{:keys [container capture-logs?]} container-config]
     (.start container)
     (-> container-config
         (assoc :id (.getContainerId container))
+        (assoc :logs (when capture-logs? (capture-logs container)))
```

Comment:
This would mean, that `nil` is associated to `:logs` if the flag is not set, wouldn't it?

Reply:
yes, that is right. would that be okay? or, perhaps using `if` with a `false` else?

Reply:
I would prefer that the key was just not present at all, that's how it is also done with e.g. network. Then the behavior would be more unified

Reply:
Hi, updated to now only add logs if requested.